src/dftbp/dftbplus/initprogram.F90

```diff
@@ -1839,12 +1839,9 @@ subroutine initProgramVariables(this, input, env)
     else
       this%tCasidaForces = .false.
     end if
-    if (this%tSccCalc) then
-      this%forceType = input%ctrl%forceType
-    else
-      if (input%ctrl%forceType /= forceTypes%orig) then
-        call error("Invalid force evaluation method for non-SCC calculations.")
-      end if
+    this%forceType = input%ctrl%forceType
```

Comment:
Is this part of the changes needed for the PR, or a stray fix?

Reply:
It is needed, as tests fail without it. It was always buggy, but by accident `this%forceType` was always zero. By putting the `TDftbPlus` instance on the heap rather than on the stack the uninitialized integer became usually non-zero, causing several non-scc tests to fail.